pkg/store/store.go

```diff
@@ -354,6 +374,61 @@ func (s Store) GetSecret(namespace, name string) (*corev1.Secret, error) {
 	return secret.(*corev1.Secret), nil
 }
 
+// GetIngressClass returns the IngressClass resource named 'name', if it exists; Otherwise, it returns an error
+func (s Store) GetIngressClass(name string) (*networkingv1.IngressClass, error) {
+	p, exists, err := s.stores.IngressClass.GetByKey(name)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, ErrNotFound{fmt.Sprintf("IngressClass %v not found", name)}
+	}
+	return p.(*networkingv1.IngressClass), nil
+}
+
+// GetIngressClassParams returns the IngressClassParams resource referenced
+// by the IngressClass resource associated with this controller, if it exists.
+// Otherwise, it returns an error
+func (s Store) GetIngressClassParams() (*kongv1alpha1.IngressClassParams, error) {
+	class, exists, err := s.stores.IngressClass.GetByKey(s.ingressClass)
```

Comment:
There's a NPE in the tests on this line - please ensure that all the CI tests for this PR pass.